examples/whisper/whisper.cpp

```diff
@@ -3027,37 +3037,55 @@ struct whisper_state * whisper_init_state(whisper_context * ctx) {
 
     // conv allocator
     {
-        whisper_allocr_graph_init(state->alloc_conv, ctx->backend,
+        bool ok = whisper_allocr_graph_init(state->alloc_conv, ctx->backend,
                 [&]() {
                     return whisper_build_graph_conv(*ctx, *state, 0);
                 });
 
+        if (!ok) {
+            WHISPER_LOG_ERROR("%s: failed to init conv allocator\n", __func__);
+            delete state;
```

Comment:
Is this enough to free the resources? `whisper_free_state` could be moved to the `whisper_state` destructor to ensure that the resources are always freed. Or maybe just call `whisper_free_state` here instead of `delete`, but that's not very good C++.

Reply:
Yes, the way it is would leak stuff. Calling `whisper_free_state()` should be fine for now